### PR TITLE
Feature/boost promote

### PR DESCRIPTION
**Pull request by ue**

### CODE REVIEW

File: src/components/scaleSlider/scaleSliderStyles.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import EStyleSheet from 'react-native-extended-stylesheet';
+
+export default EStyleSheet.create({
+  active: {
+    textAlign: 'center',
+    fontSize: 17,
+    bottom: 10,
+    color: '#788187',
```
Comment (mistikk) [inline src/components/scaleSlider/scaleSliderStyles.js]: $iconColor

File: src/components/scaleSlider/scaleSliderStyles.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    flex: 1,
+    textAlignVertical: 'center',
+    textAlign: 'center',
+    fontWeight: 'normal',
+    color: '#bdc3c7',
+  },
+  line: {
+    width: 14,
+    height: 14,
+    backgroundColor: '#357ce6',
```
Comment (mistikk) [inline src/components/scaleSlider/scaleSliderStyles.js]: $primaryBlue

File: src/components/scaleSlider/scaleSliderStyles.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    bottom: -20,
+  },
+  marker: {
+    width: 35,
+    height: 35,
+    borderRadius: 17,
+    backgroundColor: 'white',
+    borderWidth: 1,
+    position: 'absolute',
+    borderColor: '#357ce6',
```
Comment (mistikk) [inline src/components/scaleSlider/scaleSliderStyles.js]: $primaryBlue

File: src/components/scaleSlider/scaleSliderStyles.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    flexDirection: 'row',
+    alignItems: 'center',
+    justifyContent: 'space-between',
+    bottom: -20,
+  },
+  marker: {
+    width: 35,
+    height: 35,
+    borderRadius: 17,
+    backgroundColor: 'white',
```
Comment (mistikk) [inline src/components/scaleSlider/scaleSliderStyles.js]: $white

File: src/components/scaleSlider/scaleSliderView.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+import MultiSlider from '@ptomasroos/react-native-multi-slider';
+
+import styles from './scaleSliderStyles';
+
+export default class ScaleSliderView extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      activeValue: props.activeValue || props.values[0],
+      activeIndex: props.values.indexOf(props.activeValue) || 0,
```
Comment (mistikk) [inline src/components/scaleSlider/scaleSliderView.js]: can you add null check for props.values

Comment (mistikk) [reply src/components/scaleSlider/scaleSliderView.js]: also you can use lodash get

File: src/components/scaleSlider/scaleSliderView.js [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+    return (
+      <View>
+        <View style={[styles.column, { marginLeft: LRpadding, marginRight: LRpadding }]}>
+          {this._renderScale()}
+        </View>
+        <View style={styles.container}>
+          <MultiSlider
+            trackStyle={{ backgroundColor: '#bdc3c7' }}
+            selectedStyle={{ backgroundColor: '#357ce6' }}
```
Comment (mistikk) [inline src/components/scaleSlider/scaleSliderView.js]: you can move to style file if it not necessary in here

File: src/components/textInput/view/textInputStyles.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
+import EStyleSheet from 'react-native-extended-stylesheet';
+
+export default EStyleSheet.create({
+  optionsWrapper: {
+    backgroundColor: 'white',
```
Comment (mistikk) [inline src/components/textInput/view/textInputStyles.js]: $white

File: src/components/textInput/view/textInputStyles.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+    borderTopWidth: 0,
+    borderColor: '#000',
```
Comment (mistikk) [inline src/components/textInput/view/textInputStyles.js]: $black

File: src/components/textInput/view/textInputView.js [hunk 9]
```diff
@@ -1,9 +1,19 @@
...
+import React, { Fragment } from 'react';
 import { TextInput } from 'react-native';
 import { connect } from 'react-redux';
 
+// Styles
+import styles from './textInputStyles';
+
 const TextInputView = ({ isDarkTheme, innerRef, ...props }) => (
-  <TextInput ref={innerRef} keyboardAppearance={isDarkTheme ? 'dark' : 'light'} {...props} />
+  <Fragment>
```
Comment (mistikk) [inline src/components/textInput/view/textInputView.js]: why did you put fragment?

File: src/containers/pointsContainer.js [hunk 10]
```diff
@@ -1,20 +1,23 @@
+/* eslint-disable no-unused-vars */
```
Comment (mistikk) [inline src/containers/pointsContainer.js]: please remove

File: src/containers/pointsContainer.js [hunk 11]
```diff
@@ -90,7 +112,8 @@ class PointsContainer extends Component {
 
     await getUser(username)
       .then(userPoints => {
-        this.setState({ userPoints });
+        const balance = Math.round(get(userPoints, 'points') * 1000) / 1000;
+        this.setState({ userPoints, balance });
       })
       .catch(err => {
         Alert.alert(err);
```
Comment (mistikk) [inline src/containers/pointsContainer.js]: err.message

File: src/containers/pointsContainer.js [hunk 12]
```diff
@@ -114,6 +137,17 @@ class PointsContainer extends Component {
...
   };
 
+  _getUserBalance = async username => {
+    await getUser(username)
+      .then(userPoints => {
+        const balance = Math.round(get(userPoints, 'points') * 1000) / 1000;
+        return balance;
+      })
+      .catch(err => {
+        Alert.alert(err);
```
Comment (mistikk) [inline src/containers/pointsContainer.js]: err.message

File: src/containers/pointsContainer.js [hunk 13]
```diff
@@ -135,6 +169,26 @@ class PointsContainer extends Component {
...
+        this.setState({ isLoading: false });
+        dispatch(toastNotification(intl.formatMessage({ id: 'alert.successful' })));
+        navigation.goBack();
+      })
+      .catch(error => {
+        Alert.alert(
+          `Fetching data from server failed, please try again or notify us at info@esteem.app \n${error.message.substr(
+            0,
+            20,
+          )}`,
```
Comment (mistikk) [inline src/containers/pointsContainer.js]: can you get error message from language file

File: src/providers/esteem/ePoint.js [hunk 14]
```diff
@@ -1,9 +1,9 @@
 import { Alert } from 'react-native';
-import { Client, PrivateKey } from 'dsteem';
+// import { Client, PrivateKey } from 'dsteem';
```
Comment (mistikk) [inline src/providers/esteem/ePoint.js]: please remove if it is not necessary

File: src/providers/esteem/ePoint.js [hunk 15]
```diff
@@ -1,9 +1,9 @@
...
 import ePointApi from '../../config/ePoint';
 
 // Utils
-import { decryptKey } from '../../utils/crypto';
+// import { decryptKey } from '../../utils/crypto';
```
Comment (mistikk) [inline src/providers/esteem/ePoint.js]: please remove if it is not necessary

File: src/screens/promote/screen/promoteScreen.js [hunk 16]
```diff
@@ -0,0 +1,280 @@
+/* eslint-disable prefer-destructuring */
+/* eslint-disable no-unused-expressions */
+/* eslint-disable no-undef */
+/* eslint-disable no-return-assign */
```
Comment (mistikk) [inline src/screens/promote/screen/promoteScreen.js]: :D

File: src/screens/promote/screen/promoteScreen.js [hunk 17]
```diff
@@ -0,0 +1,280 @@
...
+          isLoading,
+          accounts,
+          currentAccountName,
+          balance: _balance,
+          promote,
+          currentAccount,
+          getUserDataWithUsername,
+        }) => (
+          <Fragment>
+            <BasicHeader title="Promote" />
```
Comment (mistikk) [inline src/screens/promote/screen/promoteScreen.js]: can you get string from language file

File: src/screens/promote/screen/promoteScreen.js [hunk 18]
```diff
@@ -0,0 +1,280 @@
...
+                          <TextInput
+                            style={styles.input}
+                            onChangeText={text => this._handleOnPermlinkChange(text)}
+                            value={permlink}
+                            placeholder={intl.formatMessage({ id: 'promote.permlink' })}
+                            placeholderTextColor="#c1c5c7"
+                            autoCapitalize="none"
+                          />
+                        )}
+                        renderItem={({ item, i }) => (
```
Comment (mistikk) [inline src/screens/promote/screen/promoteScreen.js]: can you remove i

File: src/screens/promote/screen/promoteScreen.js [hunk 19]
```diff
@@ -0,0 +1,280 @@
...
+                          >
+                            <Text style={styles.autocomplateItemText}>{item}</Text>
+                          </TouchableOpacity>
+                        )}
+                      />
+                    </View>
+                  </Fragment>
+
+                  <View style={styles.total}>
+                    <Text style={styles.day}>{`${day} days `}</Text>
```
Comment (mistikk) [inline src/screens/promote/screen/promoteScreen.js]: from intl

File: src/screens/promote/screen/promoteScreen.js [hunk 20]
```diff
@@ -0,0 +1,280 @@
...
+                  >
+                    <Text style={styles.buttonText}>
+                      {intl.formatMessage({ id: 'transfer.next' })}
+                    </Text>
+                  </MainButton>
+                </View>
+              </ScrollView>
+            </View>
+            <ActionSheet
+              ref={o => (this.ActionSheet = o)}
```
Comment (mistikk) [inline src/screens/promote/screen/promoteScreen.js]: you can use createRef function (you can look contructure method of delegateScreen.js)

**Comment by ue** [comment]
@mistikk updated

**Comment by mistikk** [comment]
I think you didnt see all of comment, can you check again @ue

File: src/containers/pointsContainer.js (hunk 13 above)
Comment (ue) [reply src/containers/pointsContainer.js]: that is a different process its need another effort

**Comment by ue** [comment]
@mistikk fixed

**Pull request merged**